### PR TITLE
fix(sso): fixes MGDSTRM-8776, fixes client secret reset check for existing account

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
fix(sso): fixes MGDSTRM-8776, fixes client secret reset check for existing account

## Verification Steps
Existing connector integration tests should continue to pass. Locally tested using redhat sso config as outlined in https://issues.redhat.com/browse/MGDSTRM-8191.

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
~~- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)~~
~~- [ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

**Comment by dhirajsb** [comment]
@JameelB I fixed the failing connectors integration test for redhat sso in this PR. Please review and manually verify that it works in your redhat sso environment.

**Review by machi1990 — APPROVED** [review]
This looks good to me and a very good catch. 
I'd love for @JameelB , @ziccardi and @akoserwal to have a look as well.

File: pkg/services/sso/redhatsso_service.go
Comment (JameelB) [inline pkg/services/sso/redhatsso_service.go:164]: Is this correct? Do we regenerate the secret everytime we get it? @ziccardi @akoserwal

Comment (ziccardi) [reply pkg/services/sso/redhatsso_service.go:164]: @JameelB My memory can be wrong, but I believe the reason behind that was that on redhatsso the only way to access the secret is by regenerating it (something similar to what happens with github tokens), so we decided that, for the moment, we always regenerate it. However I think it can be removed now. I leave the final word to @akoserwal

Comment (JameelB) [reply pkg/services/sso/redhatsso_service.go:164]: Don't think we use this in the internal kafka services but I see it being used in connectors still: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/connector/internal/services/connector_cluster.go#L936

**Review by JameelB — APPROVED** [review]
Verified this locally by running integration tests using rhsso client. All tests are passing for both Kafka and Connectors.

File: pkg/services/sso/redhatsso_service.go
Comment (dhirajsb) [reply pkg/services/sso/redhatsso_service.go:164]: Yeah, the method name doesn't make it obvious that the secret is actually regenerated as there is no way to retrieve it in redhat sso. That's what we are using it for in connectors.

Comment (akoserwal) [reply pkg/services/sso/redhatsso_service.go:164]: make sense.

**Comment by machi1990** [comment]
Hi @dhirajsb looks like CI is not happy after this was merged: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/runs/6814346205?check_suite_focus=true

Can you have a look?

**Comment by dhirajsb** [comment]
@machi1990 the error is happening in an unrelated place:
```
[dbokde@dbokde kas-fleet-manager (main =)]$ make lint 
/home/dbokde/projects/bf2f/kas-fleet-manager/bin/golangci-lint run \
	./cmd/... \
	./pkg/... \
	./internal/... \
	./test/...
internal/kafka/internal/handlers/admin_kafka_test.go:74:20: response body must be closed (bodyclose)
			Expect(rw.Result().StatusCode).To(Equal(tt.wantStatusCode))
			                ^
internal/kafka/internal/handlers/admin_kafka_test.go:161:20: response body must be closed (bodyclose)
			Expect(rw.Result().StatusCode).To(Equal(tt.wantStatusCode))
			                ^
internal/kafka/internal/handlers/admin_kafka_test.go:214:20: response body must be closed (bodyclose)
			Expect(rw.Result().StatusCode).To(Equal(tt.wantStatusCode))
			                ^
make: *** [Makefile:281: lint] Error 1
```

**Comment by ziccardi** [comment]
Probably a PR has been merged before the linter was introduced.

I already discussed that with @pawelpaszki .

#1091 fixes those issues

**Pull request merged**
